doc/sphinxext/credit_tools.py

```diff
@@ -84,6 +126,18 @@ def generate_credit_rst(app=None, *, verbose=False):
 
     unknown_emails: set[str] = set()
 
+    # Use doc/changes/*.rst to map PR numbers to names
```

Comment:
a bit concerned about this, as (1) authors listed in changelog won't necessarily match those in `Co-authored-by` commit trailers, and (2) for older changelog entries, we must take in on faith that the PR number was entered correctly.